Madness/Repetition.swift

```diff
@@ -15,6 +15,16 @@ public func sepBy<C: CollectionType, T, U>(parser: Parser<C, T>.Function, _ sepa
 	return sepBy1(parser, separator) <|> pure([])
 }
 
+/// Parses 1 or more `parser` ended by `separator`.
+public func endBy1<C: CollectionType, T, U>(parser: Parser<C, T>.Function, _ separator: Parser<C, U>.Function) -> Parser<C, [T]>.Function {
+	return some(parser <* separator)
+}
+
+/// Parses 0 or more `parser` ended by `separator`.
+public func endBy<C: CollectionType, T, U>(parser: Parser<C, T>.Function, _ separator: Parser<C, U>.Function) -> Parser<C, [T]>.Function {
+	return many(parser <* separator)
+}
```

Comment:
> If I’m understanding this correctly, endBy can be a convenient way to handle lists with terminators: Is that correct?

I think too.

> If so, perhaps we should rename the `separator` parameter to `terminator` instead. What do you think?

I selected `separator` because of two reasons.
- same as `sepBy/sepBy1`
- stealing names from [Parsec](https://hackage.haskell.org/package/parsec-3.1.9/docs/Text-Parsec-Combinator.html) and [Trifecta](https://hackage.haskell.org/package/parsers-0.12.3/docs/Text-Parser-Combinators.html)

But, that's better to understand `endBy` :+1: